lib/manager/npm/extract/index.ts

```diff
@@ -237,6 +237,10 @@ export async function extractPackageFile(
     }
     if (dep.currentValue.startsWith('file:')) {
       dep.skipReason = SkipReason.File;
+      // https://github.com/npm/cli/issues/1432
+      // Explanation:
+      //  - npm install --package-lock-only is buggy for transitive deps in file: references
+      //  - So we set artifactUpdateApproach to false if file: refs are found *and* the user hasn't explicitly set the value already
       hasFileRefs = true;
```

Comment:
https://github.com/renovatebot/renovate/commit/07affef41c6b4c7cb7d0413467800bdde4eb9876 is missing here